pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -286,6 +288,14 @@ public void setCustomProperties(Map<String, String> properties) {
     _customProperties.putAll(properties);
   }
 
+  private void setDateTimeFormatSpec(DateTimeFormatSpec formatSpec) {
```

Comment:
Let's make this api public to replace the current `setSimpleDateFormat()`. To keep consistent behavior, we should not allow setting just the SDF